framework/src/modules/chain/blocks/chain.js

```diff
@@ -141,16 +141,48 @@ class BlocksChain {
 		if (lastBlock.height === 1) {
 			throw new Error('Cannot delete genesis block');
 		}
-		const previousBlock = await popLastBlock(
-			this.storage,
-			this.interfaceAdapters,
-			this.genesisBlock,
-			this.roundsModule,
-			this.slots,
-			lastBlock,
-			this.exceptions
+		return this.storage.entities.Block.begin('Chain:deleteBlock', async tx => {
+			const previousBlock = await popLastBlock(
+				this.storage,
+				this.interfaceAdapters,
+				this.genesisBlock,
+				this.roundsModule,
+				this.slots,
+				lastBlock,
+				this.exceptions,
+				tx
+			);
+			return previousBlock;
+		});
+	}
+
+	async deleteLastBlockAndStoreInTemp(lastBlock) {
+		if (lastBlock.height === 1) {
+			throw new Error('Cannot delete genesis block');
+		}
+		return this.storage.entities.Block.begin(
+			'Chain:deleteBlockAndStoreInTemp',
+			async tx => {
+				const previousBlock = await popLastBlock(
+					this.storage,
+					this.interfaceAdapters,
+					this.genesisBlock,
+					this.roundsModule,
+					this.slots,
+					lastBlock,
+					this.exceptions,
+					tx
+				);
+				const parsedDeletedBlock = parseBlockToJson(lastBlock);
```

Comment:
You returned `previousBlock` but to store used `lastBlock`, any particular reason for it?

Reply:
I think `blockToJSON` or `convertBlockToJSON` suits as a better name, as `parse` is only meant to describe the act of reading and understanding the contents of the block

Reply:
@nazarhussain I am keeping the same interface as `deleteLastBlock` since soon we might be replacing the functionality of deleteLastBlock with thiis function.

Reply:
@limiaspasdaniel what do you mean? I don't understand your point?

Reply:
Ignore my comment sorry, I read the name again and it indeed makes sense :)